### DISCUSSION TITLE
Scripting needs a way to be able to be organized.

### DISCUSSION THREAD

**Discussion by MarceloGToonzzz** [godotengine/godot-proposals#4011] [Editor]
I think that you should be able to group scripts so the script tab can be more organized. Like this.
![image](https://user-images.githubusercontent.com/68459085/154779366-1bf29f6d-1fb6-4720-b150-fd2ea270ae35.png)
You would be able to name the folder to make categories.

**Comment by seppoday** [comment]
How about categories from folder destination of script? That might be good because user don't need to build categories. Scripts are categorized cleanly automatically and you don't need to remember to move script from category to category if you move it in folders...etc.

Although it might be bad for someone that keeps all scripts under one folder.

Also all of them can be collapsable and there is on for documentation pages :)

![Categories from folders](https://user-images.githubusercontent.com/62170071/154813942-260caee5-c759-49f7-996a-3a7e23f0e6d3.png)

**Reply by Gallilus** [reply]
The only thing that is missing here is a way to prevent every script from being in its own category.

> ../Enemies
> enemy.gd
> enemy_tree.gd

instead of 

> ../Enemies/EnemyOne
> enemy.gd
> ../Enemies/EnemyTree
> enemy_tree.gd

**Reply by xix-xeaon** [reply]
I don't think I'd want to lose part of the path. But the path part could be faded out / smaller text etc. I wouldn't want the full path though, categories are nice.
> ../Enemies
> EnemyOne/  enemy.gd
> EnemyTree/  enemy_tree.gd

**Reply by Gallilus** [reply]
Also here we are assuming a certain file structure from users. So maybe not the best id for Core, as a plugin maybe?

**Reply by aXu-AP** [reply]
Godot does support this to some extent with option to sort and list scripts with path name (Editor Settings / Text Editor / Script List / Sort Scripts By, List Script Names As). Altough this doesn't make collapsable structure.

**Reply by aXu-AP** [reply]
> The only thing that is missing here is a way to prevent every script from being in its own category.

Yeah, there are many different ways project could be organized. I have a lot of folders where is one script, one scene and other assets.
One possibility to this is, that there would be an editor setting as to how deep these path separators go. Another way would be setting a minimum count of scripts in a category. Kind of a "smart/automatic" categories.

**Comment by omniuni** [comment]
One simple option would be "Display open Scripts as Tree" which would literally just show them in a file tree.

**Reply by aXu-AP** [reply]
I like @seppoday's idea better, because not every directory contains scripts. Also tree requires more horizontal space.

**Reply by omniuni** [reply]
The IDEA IDE's also have a way to improve that, by collapsing empty middle layers. Although not necessarily perfectly ideal, part of why I recommended this is simply because trees already exist and require fairly little logic to handle edge cases such as nested folders.
